src/Forms/AutoBreakUnbreakLines.cs

```diff
@@ -181,17 +181,38 @@ private void AddToListView(Paragraph p, string newText)
 
         private bool IsFixAllowed(Paragraph p)
         {
-            foreach (ListViewItem item in listViewFixes.Items)
+            var count = listViewFixes.Items.Count;
+            int index = (count / 2) - 1;
+            index = index >= 0 ? index : 0;
+
+            var pr = listViewFixes.Items[index].Tag as Paragraph;
+            if (pr.ID == p.ID)
+                return listViewFixes.Items[index].Checked;
+
+            if (pr.Number > p.Number)
             {
-                if (item.Tag.ToString() == p.ToString())
-                    return item.Checked;
+                for (int i = 0; i < index; i++)
+                {
+                    var pp = listViewFixes.Items[i].Tag as Paragraph;
+                    if (pp.ID == p.ID)
+                        return listViewFixes.Items[i].Checked;
+                }
+            }
+            else
+            {
+                for (int i = index; i < count; i++)
+                {
+                    var pp = listViewFixes.Items[i].Tag as Paragraph;
+                    if (pp.ID == p.ID)
+                        return listViewFixes.Items[i].Checked;
+                }

```

Comment:
:+1: 
The second list  (_paragraphs2) would not be necessary, if you put that loop in ButtonOk_Click().